### ISSUE TITLE
oxfmt rewrites /*@__PURE__*/ outside parens, breaking rolldown tree-shaking

### ISSUE THREAD

**Issue by jdalton** [oxc-project/oxc#22797]
## Summary

oxfmt rewrites `/*@__PURE__*/` (and related rolldown/rollup/terser pure-annotation comments) into a position where rolldown silently ignores them. The reformatted output is valid JavaScript but loses its tree-shaking semantics.

## Reproduce

Input (oxfmt 1.x, current main):

```ts
function getMatchesGlob() {
  if (!probed) {
    const fn = (
      /*@__PURE__*/ require('node:path') as typeof import('node:path') & {
        matchesGlob?: unknown
      }
    ).matchesGlob
    return fn
  }
}
```

Run with the minimal config:

```json
{
  "tabWidth": 2,
  "printWidth": 80,
  "singleQuote": true,
  "trailingComma": "all",
  "semi": false
}
```

Output (oxfmt rewrites the comment outside the parens):

```ts
function getMatchesGlob() {
  if (!probed) {
    const fn = /*@__PURE__*/ (
      require('node:path') as typeof import('node:path') & {
        matchesGlob?: unknown
      }
    ).matchesGlob
    return fn
  }
}
```

## Expected

The `/*@__PURE__*/` comment should stay adjacent to the `require()` call expression (i.e. immediately to its left, inside the parens). When it gets hoisted outside the parenthesized cast, rolldown's pure-annotation detector treats the parenthesized expression as the annotation target (not the inner call) and silently drops the annotation. The call ends up not marked pure, and dead-code elimination misses it.

## Why this matters

The rolldown docs are explicit about position requirements: https://rolldown.rs/in-depth/dead-code-elimination#pure. The same is true for rollup, terser, and esbuild — all use position-adjacent matching for the magic-comment family.

Affected annotations (all positional):

- `/*@__PURE__*/` and `/* @__PURE__ */`
- `/*#__PURE__*/` and `/* #__PURE__ */`
- `/*@__NO_SIDE_EFFECTS__*/` and `/* @__NO_SIDE_EFFECTS__ */`
- `/*#__NO_SIDE_EFFECTS__*/` and `/* #__NO_SIDE_EFFECTS__ */`

## Workaround

Restructure to an intermediate const so the cast sits on its own statement:

```ts
const pathMod = /*@__PURE__*/ require('node:path') as typeof import('node:path') & {
  matchesGlob?: unknown
}
const fn = pathMod.matchesGlob
```

oxfmt handles this idempotently — line-reflows the cast but doesn't move the comment.

## Suggested fix

In oxfmt's comment-attachment phase, treat the magic-comment family above as anchored to the immediately-following AST node (a CallExpression in the `/*@__PURE__*/` case, a FunctionDeclaration / ClassDeclaration in the `/*@__NO_SIDE_EFFECTS__*/` case). Skip the usual leading-comment-hoisting rule for those.

## Repro environment

- oxfmt: latest (reproduced locally on 2026-05-28)
- Node: 26.1.0
- Platform: darwin-arm64

Happy to send a PR with a regression test + the comment-anchor patch if the maintainers agree on the approach.

**Comment by camc314** [comment]
duplicate of #22795

**Comment by jdalton** [comment]
😮‍💨 Sorry @camc314! Thank you for the housekeeping!
